clash-lib/src/Clash/Normalize/Transformations/Case.hs

```diff
@@ -263,29 +263,38 @@ caseCon' ctx@(TransformContext is0 _) e@(Case subj ty alts) = do
       (Literal l,_,_) -> caseCon ctx1 (Case (Literal l) ty alts)
       -- WHNF of subject is a data-constructor, try `caseCon` with that
       (Data _,_,_) -> caseCon ctx1 (Case subj1 ty alts)
+      -- WHNF of subject is _|_, in the form of `error`: that means that the
+      -- entire case-expression is evaluates to _|_
+      (Prim pInfo,repTy:_:callStack:msg:_,ticks)
+        | primName pInfo == "GHC.Err.error" ->
+        let e1 = mkApps (mkTicks (Prim pInfo) ticks)
+                        [repTy,Right ty,callStack,msg]
+         in changed e1
       -- WHNF of subject is _|_, in the form of `absentError`: that means that
       -- the entire case-expression is evaluates to _|_
       (Prim pInfo,_:msgOrCallStack:_,ticks)
         | primName pInfo == "Control.Exception.Base.absentError" ->
         let e1 = mkApps (mkTicks (Prim pInfo) ticks)
                         [Right ty,msgOrCallStack]
         in  changed e1
-      -- WHNF of subject is _|_, in the form of `absentError`, `patError`,
-      -- or `undefined`: that means the entire case-expression is _|_
+      -- WHNF of subject is _|_, in the form of  `patError`, `undefined`, or
+      -- `errorWithoutStackTrace`: that means the entire case-expression is _|_
       (Prim pInfo,repTy:_:msgOrCallStack:_,ticks)
         | primName pInfo `elem` ["Control.Exception.Base.patError"
-                                ,"GHC.Err.undefined"] ->
+                                ,"GHC.Err.undefined"
+                                ,"GHC.Err.errorWithoutStackTrace"] ->
         let e1 = mkApps (mkTicks (Prim pInfo) ticks)
                         [repTy,Right ty,msgOrCallStack]
         in  changed e1
       -- WHNF of subject is _|_, in the form of our internal _|_-values: that
       -- means the entire case-expression is _|_
       (Prim pInfo,[_],ticks)
         | primName pInfo `elem` [ Text.showt 'NP.undefined
-                                , "Clash.GHC.Evaluator.undefined"
-                                , "EmptyCase"] ->
+                                , Text.showt 'NP.undefinedX ] ->
         let e1 = mkApps (mkTicks (Prim pInfo) ticks) [Right ty]
         in changed e1
+      -- WHNF of subject is _|_, in the form of `errorX`: that means that
+      -- the entire case-expression is evaluates to _|_
```

Comment:
This hasn't changed, but reading this case back it seems odd that we silently discard the ticks when it's `errorX` since we keep the ticks for other forms of error. Is this intentional?

Reply:
Seems like an oversight when the code was originally introduced https://github.com/clash-lang/clash-compiler/commit/cedd0eeafe051e515f17403bf96e65ff979f0f54, I'll fix it.